joomla/joomla-cms
# [5.0] Work around a successful upgrade with silent errors

### Summary of Changes

This is the same as 
- https://github.com/joomla/joomla-cms/pull/41367

But for 5.0 branch.
I think upmerge will be complicated, so I have re-apply changes.


### Testing Instructions
If that https://github.com/joomla/joomla-cms/pull/41367 will be tested, then this one also should be good.

Note: The patch https://github.com/joomla/joomla-cms/pull/41367 should be applied on 4.x version, before test.

**Test normal**
Try run update from 4.4 => 5.0
The update should work as usual.

**Test with error**
Simulate an error, add an exception `throw new Exception('Test error');` somehwere here https://github.com/joomla/joomla-cms/blob/7eff1b78215743a1f4002c9729dcd17e2aa35d4d/administrator/components/com_admin/script.php#L2557-L2565
And do update.

**Actual:** The update finished with Success message. And nothing in logs.
**Expected:** The update finished with error message. The error is logged.


### Link to documentations
Please select:
